File: Php/app/Urls/CashgameLeaderboardUrlModel.php
<?php
namespace app\Urls{

	use app\RouteFormats;
	use app\Urls\BaseClasses\HomegameYearUrlModel;
	use entities\Homegame;

	class CashgameLeaderboardUrlModel extends HomegameYearUrlModel{

		public function __construct(Homegame $homegame, $year){
			parent::__construct(RouteFormats::cashgameLeaderboard, RouteFormats::cashgameLeaderboardWithYear, $homegame, $year);
		}

	}

}

File: Web/Models/UrlModels/CashgameLeaderboardUrlModel.cs
using Core.Classes;
using Web.Routing;

namespace Web.Models.UrlModels{

	public class CashgameLeaderboardUrlModel : HomegameYearUrlModel{

		public CashgameLeaderboardUrlModel(Homegame homegame, int? year)
            : base(RouteFormats.CashgameLeaderboard, RouteFormats.CashgameLeaderboardWithYear, homegame, year)
        {
		}

	}

}